cirq_superstaq/ops/qubit_gates.py

```diff
@@ -753,6 +753,75 @@ def _from_json_dict_(cls, **kwargs: Any) -> "IXGate":
 AQTICCX = AQTITOFFOLI = IX.controlled(2, [0, 0])
 
 
+@cirq.value_equality(approximate=True)
+class StrippedCZGate(cirq.Gate):
+    """The Stripped CZ Gate is the gate that is actually being performed by Hilbert,
+    which is just a regular CZ gate when phi = 0, and is corrected into a CZ
+    gate by RZ gates afterwards if phi is nonzero.
+    """
+
+    def __init__(self, phi: cirq.TParamVal = 0) -> None:
+        """Args: phi: CZ-rotation angle in radians"""
+        self.phi = phi
+
+    def _num_qubits_(self) -> int:
+        return 2
+
+    def _unitary_(self) -> Optional[npt.NDArray[np.complex_]]:
+        if self._is_parameterized_():
+            return None
+        return np.diag(
+            [1.0, np.exp(1j * self.phi), np.exp(1j * self.phi), np.exp(1j * (2 * self.phi - np.pi))]
+        )
+
+    def _value_equality_values_(self) -> cirq.TParamVal:
+        if cirq.is_parameterized(self.phi):
+            return self.phi
+        return self.phi % (2 * np.pi)
+
+    def _value_equality_approximate_values_(self) -> cirq.PeriodicValue:
+        return cirq.PeriodicValue(self.phi, 2 * np.pi)
+
+    def __pow__(self, exponent: cirq.TParamVal) -> "StrippedCZGate":
+        if exponent == 1:
+            return StrippedCZGate(exponent * self.phi)
```

Comment:
is it worth adding the inverse here? we can use:
```suggestion
        if exponent == 1:
            return StrippedCZGate(self.phi)
        elif exponent == -1:
            return StrippedCZGate(-self.phi)
```
(also removed the `exponent *` because the exponent is 1)

Reply:
We may as well do arbitrary integer exponents at this point, with something like
```python
if exponent == 0:
    return cirq.IdentityGate(2)
elif exponent == int(exponent):
    new_phi = exponent * phi
    if exponent % 2 == 1:
        return cirq.StrippedCZGate(new_phi)
    return cirq.ParallelGates(cirq.Rz(new_phi), cirq.Rz(new_phi))
return NotImplemented
```

Reply:
Also, is `Gate.__pow__` allowed to return a sequence of gates?  If so, we could also do non-integer powers.

Reply:
Actually: is `Gate.__pow__` allowed to return a sequence of gates?  If so, maybe we should return a `CZPowGate` + `ParallelGates` for non-integer exponents.

Reply:
nice. to preserve global phase in the fractional/DiagonalGate case i think it has to be (definitely double-check):
```python
phases = [np.exp(-1j * new_phi), 1, 1, np.exp(1j * (new_phi - exponent * np.pi))]
```
though i might also be tempted to skip this case and only support integer powers

(also i think multiple-of-two case should be `cirq.ParallelGate(cirq.rz(new_phi), 2)`)

Reply:
> Actually: is `Gate.__pow__` allowed to return a sequence of gates?

i don't think so

Reply:
For global phase, the phases I had are correct: the |00> state is completely unaffected by the gate.  So actually the `Rz`s I had should be `ZPowGate`s.

Reply:
out of curiosity what do you mean by "correct"? physically aren't the two gates the same? (obviously nothing wrong with with the ZPowGate version, just interested)

Reply:
oh wait does it have to do with the rydberg level?

Reply:
I meant "correct" in the sense of getting the global phase correct as well.  The |00> state is completely unaffected by the pulses that implement the CZ gate (which couples only the |1> state to the rydberg level), so the <00|gate|00> should be 1.